lib/switchtec.c

```diff
@@ -133,9 +198,140 @@ struct switchtec_dev *switchtec_open(const char *device)
 	else
 		errno = ENODEV;
 
+	ret->device_id = ret->ops->get_device_id(ret);
```

Comment:
get_device_id() isn't defined anywhere yet so this will likely crash the program.

Either patches that define these for all architectures need to come before this patch, or you need to check if get_device_id is NULL or not.